x-pack/plugins/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -345,6 +347,67 @@ export class ObservabilityAIAssistantClient {
     });
   };
 
+  get_dataset_info = async (
+    index: string
+  ): Promise<{
+    indices: string[];
+    fields: Array<{ name: string; description: string; type: string }>;
+  }> => {
+    // if empty dataview, get list of all dataviews
+
+    let indices: string[] = [];
+
+    try {
+      const body = await this.dependencies.esClient.indices.resolveIndex({
+        name: index === '' ? '*' : index,
+        expand_wildcards: 'open',
+      });
+      indices = body.indices.map((i) => i.name);
+    } catch (e) {
+      indices = [];
+    }
+
+    if (index === '') {
+      return {
+        indices,
+        fields: [],
+      };
+    }
+
+    if (indices.length === 0) {
+      try {
+        const body = await this.dependencies.esClient.indices.resolveIndex({
+          name: '*',
+          expand_wildcards: 'open',
+        });
+        indices = body.indices.map((i) => i.name);
+      } catch (e) {
+        indices = [];
+      }
+
+      return {
+        indices,
+        fields: [],
+      };
+    }
+
+    const fields = await this.dependencies.dataviews.getFieldsForWildcard({
+      pattern: index,
+    });
+
+    // else get all the fields for the found dataview
+    return {
```

Comment:
can you return this in a more compact format? e.g. CSV?

Reply:
seems to me json is correct format for kibna client/server communication. maybe we want to compress this down before sending it out to LLM ? i'll give it a try

Reply:
seems to work fine